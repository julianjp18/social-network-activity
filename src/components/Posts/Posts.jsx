import React from 'react';
import { Avatar, Image, List, Space } from "antd";
import { Link } from "react-router-dom";
import { LikeOutlined, MessageOutlined, StarOutlined } from "@ant-design/icons";

const IconText = ({ icon, text }) => (
  <Space>
    {React.createElement(icon)}
    {text}
  </Space>
);

const Posts = ({ allComments, posts }) => {

  const getTotalCommentsNumber = (postId) => {
    let totalComments = 0;
    allComments?.forEach((comment) => {
      if (comment.postId === Number.parseInt(postId))
        totalComments++;
    });
    
    return totalComments;
  };

  return (
    <List
      itemLayout="vertical"
      size="large"
      pagination={{
        onChange: (page) => {
          console.log(page);
        },
        pageSize: 5,
      }}
      dataSource={posts}
      footer={
        <div>
          Made by <b>Julian Perez</b> for Patagonian
        </div>
      }
      renderItem={(item) => (
        <List.Item
          key={item.title}
          actions={[
            <IconText
              icon={StarOutlined}
              text="156"
              key="list-vertical-star-o"
            />,
            <IconText
              icon={LikeOutlined}
              text="156"
              key="list-vertical-like-o"
            />,
            <Link to={`/comments/${item.id}`}>
              <IconText
                icon={MessageOutlined}
                text={getTotalCommentsNumber(item.id)}
                key="list-vertical-message"
              />
            </Link>
            ,
          ]}
          extra={
            <Image
              width={150}
              src="https://picsum.photos/1200"
            />
          }
        >
          <List.Item.Meta
            avatar={<Avatar src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png" />}
            title={
              <Link to={`/comments/${item.id}`}>
                {item.title}
              </Link>
            }
            description={item.description}
          />
          {item.body}
        </List.Item>
      )}
    />
  );
};

export default Posts;
