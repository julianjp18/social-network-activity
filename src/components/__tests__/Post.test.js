import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Posts from '../Posts/Posts';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() })

describe('Posts component tests', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Posts />)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});