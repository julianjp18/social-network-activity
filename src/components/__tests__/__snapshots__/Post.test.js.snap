// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component tests renders correctly 1`] = `
<List
  footer={
    <div>
      Made by 
      <b>
        Julian Perez
      </b>
       for Patagonian
    </div>
  }
  itemLayout="vertical"
  pagination={
    Object {
      "onChange": [Function],
      "pageSize": 5,
    }
  }
  renderItem={[Function]}
  size="large"
/>
`;
